save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/ProjectView.kt

```diff
@@ -148,6 +160,8 @@ class ProjectView : AbstractView<ProjectViewProps, ProjectViewState>(false) {
             val role = getHighestRole(currentUserRole, props.currentUserInfo?.globalRole)
             setState {
                 selfRole = role
```

Comment:
we have `selfRole` -- it's a highest role already. can we use onlyit?

Reply:
No, because there may be a situation when a person in the organization is an ```ADMIN```, and the ```VIEWER``` in project, a large role will give ```ADMIN``` and allow access to the button that returns demo, but logically from projectPermissionEvaluator this should not happen

Reply:
It seems that you have `roleA`, `roleB` and `max(roleA, roleB)`. Why do you need to store `max(roleA, roleB)` then? It seems that `getHighestRole` is not that expensive so that you should not create a new state for it

Reply:
As it was explained to me, because `max(roleA, roleB)` can be calculated only when `roleA` is from user and it's `SUPER_ADMIN`, or `roleA` is organization and it's `OWNER`

Reply:
It all seems a bit wrong

Guess in this case we need `globalRole`, `projectRole`, `organizationRole`. Probably `selfRole` is not the best name here as it seems complicating